nginx/config.go

```diff
@@ -73,8 +73,9 @@ http {
 	# treated as if it were HTTPs.
 	map $http_x_forwarded_proto $tmp_access_scheme {
 		default $scheme;               # if X-Forwarded-Proto header is empty, $tmp_access_scheme will be the actual protocol used
-		"~^(.*, ?)?http$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value
-		"~^(.*, ?)?https$" "https";    # account for the possibility of a comma-delimited X-Forwarded-Proto header value
+		"~^(.*, ?)?https(,.* ?)?$" "https";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if https is present
+		"~^(.*, ?)?http(,.* ?)?$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if http is present
+		"~^[A-Za-z][A-Za-z0-9+.-]*" $http_x_forwarded_proto; # Use the value passed in the header if its a valid URI scheme
```

Comment:
yes...that can still be a private scheme internally used by someone, hence my question do we want to be more strict or loose on these things?I assume more strict and just add support for web sockets for now as those are the only things apart from http which most people will use.